src/integrationTest/java/com/hivemq/cli/commands/swarm/run/SwarmRunStopCommandIT.java

```diff
@@ -131,54 +126,53 @@ void tearDown() {
-
         // no current run
         final int execute0 = commandLine.execute(
-                "-url=http://" + swarm.getContainerIpAddress() + ":" + swarm.getMappedPort(REST_PORT)
-        );
+                "-url=http://" + swarm.getContainerIpAddress() + ":" + swarm.getMappedPort(REST_PORT));
         assertEquals(0, execute0);
         verify(out, times(1)).println("No run in progress.");
 
-
         final Mqtt5BlockingClient client = Mqtt5Client.builder().serverPort(hivemq.getMqttPort()).buildBlocking();
         client.connect();
         final Mqtt5BlockingClient.Mqtt5Publishes publishes = client.publishes(MqttGlobalPublishFilter.ALL);
         client.toAsync().subscribeWith().topicFilter("#").send();
 
-        final UploadScenarioRequest uploadScenarioRequest = new UploadScenarioRequest()
-                .scenarioType("XML").scenario(scenarioBase64).scenarioName("my-scenario");
+        final UploadScenarioRequest uploadScenarioRequest =
+                new UploadScenarioRequest().scenarioType("XML").scenario(scenarioBase64).scenarioName("my-scenario");
         final UploadScenarioResponse uploadScenarioResponse = scenariosApi.uploadScenario(uploadScenarioRequest);
 
         final StartRunRequest startRunRequest = new StartRunRequest();
-        startRunRequest.setScenarioId(uploadScenarioResponse.getScenarioId().toString());
+        final Integer scenarioId = uploadScenarioResponse.getScenarioId();
+        assertNotNull(scenarioId);
+        startRunRequest.setScenarioId(scenarioId.toString());
         runsApi.startRun(startRunRequest);
 
         // the scenario is started
         publishes.receive();
 
-
         // stop the scenario
+        //TODO: not sure why here a logger reset is necessary (local machine)
+        TestLoggerUtils.resetLogger();
         final int execute = commandLine.execute(
                 "-url=http://" + swarm.getContainerIpAddress() + ":" + swarm.getMappedPort(REST_PORT),
-                "-r" + 1
-        );
+                "-r" + 1);
         assertEquals(0, execute);
 
-        await().atMost(Duration.ofSeconds(3))
-                .until(() -> {
-                    final CommanderStateResponse commanderStatus = commanderApi.getCommanderStatus();
-                    System.out.println(commanderStatus);
-                    return "READY".equals(commanderStatus.getCommanderStatus());
-                });
+        await().atMost(Duration.ofSeconds(3)).until(() -> {
+            final CommanderStateResponse commanderStatus = commanderApi.getCommanderStatus();
+            System.out.println(commanderStatus);
+            return "READY".equals(commanderStatus.getCommanderStatus());
+        });
 
         final RunResponse run = runsApi.getRun("1");
         assertEquals("STOPPED", run.getRunStatus());
 
         // stop a run that does not exist
+        //TODO: however here a logger reset is not necessary (local machine)
+        //TestLoggerUtils.resetLogger();
```

Comment:
remove

Reply:
I left the TODOs in this test on purpose as it ran fine on Jenkins but failed on my machine. I think this could be the origin of a flaky test. Should be investigated in a followup ticket.